pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,254 @@
+package awsiid
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/organizations"
+	"github.com/aws/aws-sdk-go-v2/service/organizations/types"
+	"github.com/hashicorp/go-hclog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	orgAccountID             = "management_account_id"
+	orgAccountRole           = "assume_org_role"
+	orgAccRegion             = "management_account_region" // required for cache key
+	orgAccountStatus         = "ACTIVE"                    // Only allow node account id's with status ACTIVE
+	orgAccountListTTL        = "org_account_map_ttl"       // Cache the list of account for specific time, if not sent default will be used.
+	orgAccountDefaultListTTL = "3m"                        // pull account list after 3 minutes
+	orgAccountMinListTTL     = "1m"                        // Minimum TTL configuration to pull the org account list
+	orgAccountRetries        = 5
+)
+
+type orgValidationConfig struct {
+	AccountID      string `hcl:"management_account_id"`
+	AccountRole    string `hcl:"assume_org_role"`
+	AccountRegion  string
+	AccountListTTL string `hcl:"org_account_map_ttl"`
+}
+
+type orgValidator struct {
+	orgListAccountMap             map[string]bool
+	orgListAccountMapCreationTime time.Time
+	orgConfig                     *orgValidationConfig
+	mutex                         sync.RWMutex
+	orgAccountListCacheTTL        time.Duration // set from configuration else will be set to default
+	log                           hclog.Logger
+	retries                       int // fix number of retries before ttl is expired.
+}
+
+func newOrganizationValidationBase(config *orgValidationConfig) *orgValidator {
+	client := &orgValidator{
+		orgListAccountMap: make(map[string]bool),
+		orgConfig:         config,
+		retries:           orgAccountRetries,
+	}
+
+	return client
+}
+
+func (o *orgValidator) getRetries() int {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+	return o.retries
+}
+
+func (o *orgValidator) decrRetries() int {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+	if o.retries > 0 {
+		o.retries -= 1
+	}
+
+	return o.retries
+}
+
+func (o *orgValidator) configure(config *orgValidationConfig) error {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.orgConfig = config
+	// required for the way existing getClient function is designed
+	o.orgConfig.AccountRegion = "us-west-2"
+
+	// While doing configuration invalidate the map so we dont keep using old one.
+	o.orgListAccountMap = make(map[string]bool)
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.Internal, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
```

Comment:
it looks to me, that it is not internal, because this is a config, so maybe it is an invalid argument?